tools/sandbox/src/commands/gdb.rs

```diff
@@ -189,11 +190,11 @@ impl<S: Stepper> SingleThreadBase for RiscvGdb<'_, S> {
         let state = self.stepper.machine_state();
         regs.pc = state.hart.pc.read();
 
-        let array = state.hart.xregisters.struct_ref::<FnManagerIdent>();
-        for i in 0..31 {
-            // first is x0 == zero, only x1..=x31 are set
-            regs.x[i + 1] = array.read(i);
+        for register in <NonZeroXRegister as IntoEnumIterator>::iter() {
+            let index = register as usize;
+            regs.x[index + 1] = state.hart.xregisters.read_nz(register);
```

Comment:
```suggestion
            // first is x0 == zero, only x1..=x31 are set
            regs.x[index + 1] = state.hart.xregisters.read_nz(register);
```
Could you maintain the comment? it's helpful to explain why there's an offset